File: Sample Apps/Objective-C/Reference App/ViewControllers/ListingCell.m
//
//  ListingCell.m
//  Reference App
//
//  Created by Aaron Bratcher on 1/4/18.
//  Copyright © 2018 Liveperson. All rights reserved.
//

#import "ListingCell.h"

@interface ListingCell()

@property (weak, nonatomic) IBOutlet UILabel *name;
@property (nonatomic, copy) NSString *referenceID;
@property (nonatomic, weak) id<ListingCellDelegate> delegate;

@end

@implementation ListingCell

-(void)prepareForReuse {
	[super prepareForReuse];
	self.name.text = nil;
	self.referenceID = nil;
}

-(void)initializeWithDelegate:(id<ListingCellDelegate>)delegate {
	self.delegate = delegate;
}

-(void)showWithName:(NSString *)name referenceID:(NSString *)referenceID {
	self.name.text = [name copy];
	self.referenceID = [referenceID copy];
}

- (IBAction)chatButtonTapped:(id)sender {
	if(self.delegate)
		[self.delegate chatWithReferenceID:self.referenceID name:self.name.text];
}

@end

File: Sample Apps/Swift/Reference App/ViewControllers/ListingCell.swift
//
//  ReferenceCell.swift
//  CAO Test
//
//  Created by Aaron Bratcher on 11/27/17.
//  Copyright © 2017 Liveperson. All rights reserved.
//

import Foundation
import UIKit

protocol ReferenceTableCellDelegate {
	func chat(with listing: Listing, initialUserText: String?)
}

class ListingCell: UITableViewCell, Reusable {
	@IBOutlet weak var name: UILabel!

	var delegate: ReferenceTableCellDelegate?
	var listing: Listing?

	override func prepareForReuse() {
		self.listing = nil
		name.text = nil
	}

	func initialize(delegate: ReferenceTableCellDelegate) {
		self.delegate = delegate
	}

	func show(listing: Listing) {
		self.listing = listing
		name.text = listing.name
	}

	@IBAction func chatButtonTapped(_ sender: UIButton) {
		guard let listing = listing else { return }

		delegate?.chat(with: listing, initialUserText: listing.initialUserText)
	}
}
